src/main/java/command/order/AddOrderCommand.java

```diff
@@ -38,36 +42,82 @@ public void execute() {
         ArrayList<Medicine> medicines = Medicine.getInstance();
         Storage storage = Storage.getInstance();
 
-        String[] requiredParameters = {CommandParameters.NAME, CommandParameters.QUANTITY, CommandParameters.DATE};
-        String[] optionalParameter = {};
+        boolean nameExists = false;
+        String nameToAdd = parameters.get(CommandParameters.NAME);
+        String quantityToAdd = parameters.get(CommandParameters.QUANTITY);
+        int orderQuantity = Integer.parseInt(quantityToAdd);
+        String dateToAdd = parameters.get(CommandParameters.DATE);
```

Comment:
Perhaps this block of code should be after containsInvalidParameters() and containsInvalidParameterValues() to ensure that the variables exists and are valid?

Reply:
Fixed, thank you